core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4195,34 +4194,44 @@ public long getGlobalCheckpoint() {
         indexSettings.updateIndexMetaData(builder.build());
 
         store = createStore();
-        try (InternalEngine engine
-                 = createEngine(indexSettings, store, createTempDir(), NoMergePolicy.INSTANCE, null, seqNoServiceSupplier)) {
-            int numDocs = scaledRandomIntBetween(10, 100);
-            for (int docId = 0; docId < numDocs; docId++) {
-                ParseContext.Document document = testDocumentWithTextField();
-                document.add(new Field(SourceFieldMapper.NAME, BytesReference.toBytes(B_1), SourceFieldMapper.Defaults.FIELD_TYPE));
-                engine.index(indexForDoc(testParsedDocument(Integer.toString(docId), null, document, B_1, null)));
-                if (frequently()) {
-                    globalCheckpoint.set(randomIntBetween(
-                        Math.toIntExact(engine.seqNoService().getGlobalCheckpoint()),
-                        Math.toIntExact(engine.seqNoService().getLocalCheckpoint())));
+        engine = new InternalEngine(config(indexSettings, store, createTempDir(), NoMergePolicy.INSTANCE, null), seqNoServiceSupplier) {
```

Comment:
why did we lose the try with resources clause?

Reply:
We close the engine in tearDown but I put the try-clause back.

Reply:
I see - we reuse the same engine variable. I think it's cleaner to have it contained.